concordium-smart-contract-testing/src/invocation/impls.rs

```diff
@@ -21,12 +21,38 @@ use concordium_base::{
 };
 use concordium_smart_contract_engine::{
     v0,
-    v1::{self, trie},
+    v1::{self, trie, InvokeResponse, ReceiveContext, ReceiveInterruptedState},
     ExecResult, InterpreterEnergy,
 };
 use concordium_wasm::artifact::{self, CompiledFunction};
 use std::collections::{btree_map, BTreeMap};
 
+/// This auxiliary type is used in `invoke_entrypoint` below to keep track of
+/// the "to do list" in the form of a stack. It stores the necessary information
+/// to continue execution until all actions have been processed.
+enum Next {
```

Comment:
This should live in `invocation/types.rs` with the other types.

Reply:
It did at first.

But this is a type internal to this module so I see no reason for it living elsewhere. It is defined where it is used.

Reply:
If you notice it is a private type to this module.